mmf/utils/general.py

```diff
@@ -143,9 +143,36 @@ def get_optimizer_parameters(model, config):
     if is_parallel and hasattr(model.module, "get_optimizer_parameters"):
         parameters = model.module.get_optimizer_parameters(config)
 
+    for group in parameters:
+        group["params"] = list(group["params"])
+
+    check_unused_parameters(parameters, model, config)
+
     return parameters
 
 
+def check_unused_parameters(parameters, model, config):
+    optimizer_param_set = {p for group in parameters for p in group["params"]}
+    unused_param_names = [
+        n
+        for n, p in model.named_parameters()
```

Comment:
Would it be possible to write this in simple for loop format? It kinda hard to read as of now.

Reply:
I rewrote it in a for-loop (and tested it again).